iOS/UI/Browse/BrowseViewController.swift

```diff
@@ -232,20 +243,66 @@ extension BrowseViewController {
             let info = dataSource.itemIdentifier(for: indexPath),
             let source = SourceManager.shared.source(for: info.sourceId)
         else { return nil }
+        if tableView.isEditing { return nil } // Hide context menu when the user is reshuffling sources.
         return UIContextMenuConfiguration(identifier: nil, previewProvider: nil) { _ -> UIMenu? in
-            let action = UIAction(
-                title: NSLocalizedString("UNINSTALL", comment: ""),
-                image: UIImage(systemName: "trash")
-            ) { _ in
-                SourceManager.shared.remove(source: source)
-                self.viewModel.loadInstalledSources()
-                self.updateDataSource()
+            var actions: [UIMenuElement] = []
+            switch self.sectionIdentifier(for: indexPath.section) {
+            // Context menu items for a source in Installed section of the table
+            case .installed:
+                actions = [
+                    UIAction(
+                        title: NSLocalizedString("PIN", comment: ""),
+                        image: UIImage(systemName: "pin")
+                    ) { _ in
+                        SourceManager.shared.pin(source: source)
+                        self.viewModel.loadPinnedSources()
+                        self.updateDataSource()
+                    }, UIAction(
+                        title: NSLocalizedString("UNINSTALL", comment: ""),
+                        image: UIImage(systemName: "trash")
+                    ) { _ in
+                        SourceManager.shared.remove(source: source)
+                        self.viewModel.loadInstalledSources()
+                        self.updateDataSource()
+                    }
+                ]
+            // Context menu items for a source in Pinned section of the table
+            case .pinned:
+                actions = [
+                    UIMenu(title: "", options: .displayInline, children: [UIAction(
+                        title: NSLocalizedString("REORDER", comment: ""),
+                        image: UIImage(systemName: "shuffle")
+                    ) { _ in
+                        // Let user re-order sources inside the pinned section.
+                        tableView.setEditing(true, animated: true)
+                        self.updateNavBar()
+                    }]), UIAction(
+                        title: NSLocalizedString("UNPIN", comment: ""),
+                        image: UIImage(systemName: "pin.slash")
+                    ) { _ in
+                        // Remove source from the pinned array, recreate the installed source list and update the table.
+                        SourceManager.shared.unpin(source: source)
+                        self.viewModel.loadPinnedSources()
+                        self.updateDataSource()
+                    }, UIAction(
+                        title: NSLocalizedString("UNINSTALL", comment: ""),
```

Comment:
this action is duplicated for both, I think you could simplify the code so that you don't have to re-instantiate the same action in both branches